src/pages/address/fragments/transactions/AddressTransactions.tsx

```diff
@@ -43,40 +40,45 @@ const AddressTransactions: React.FC<Props> = (props: Props) => {
     setCurrentPage(currentPage + 1)
   }
 
-  const getTransfers = (invocations: Notification[]) =>
-    Promise.all(
-      invocations
-        .filter(
-          ({ state, event_name }) =>
-            state.length === 3 && event_name === 'Transfer',
+  function fixTransfers(tfxs: TransferDoraTS[]) {
+    return Promise.all(
+      tfxs.map(async tx => {
+        const json = await NeoRest.asset(tx.scripthash, network)
+        const { symbol, decimals } = json
+
+        const convertedAmount = convertToArbitraryDecimals(
+          Number(tx.amount),
+          Number(decimals),
         )
-        .map(async ({ contract, state }): Promise<Transfer> => {
-          const [{ value: from }, { value: to }, { value: amount }] = state
-
-          const response = await fetch(
-            `${GENERATE_BASE_URL()}/asset/${contract}`,
-          )
-
-          const json = await response.json()
-          const { symbol, decimals } = json
-
-          const convertedAmount = convertToArbitraryDecimals(
-            Number(amount),
-            decimals,
-          )
-
-          const convertedFrom = from ? byteStringToAddress(from) : 'mint'
-          const convertedTo = to ? byteStringToAddress(to) : 'burn'
-
-          return {
-            scripthash: contract,
-            from: convertedFrom,
-            to: convertedTo,
-            amount: convertedAmount,
-            symbol,
-          }
-        }),
+
+        return {
+          from: tx.from,
+          to: tx.to,
+          scripthash: tx.scripthash,
+          amount: convertedAmount,
+          symbol: symbol,
+        } as Transfer
+      }),
     )
+  }
```

Comment:
nitpicks:

1.) the name `fixTransfers` is kind of ambiguous in my opinion, maybe something like `convertTransferAmounts` would be more appropriate 


2.) This could be less verbose with destructuring:

```javascript
async function fixTransfers(tfxs: TransferDoraTS[]): Promise<Transfer[]> {
  return Promise.all(
    tfxs.map(async ({ scripthash, amount, from, to }) => {
      const { symbol, decimals } = await NeoRest.asset(scripthash, network);
      const convertedAmount = convertToArbitraryDecimals(Number(amount), Number(decimals));
      return { from, to, scripthash, amount: convertedAmount, symbol } as Transfer;
    }),
  );
}
````

Reply:
I believe it did more than just converting amounts when it had N2 code there as well. Either way, updated.